littlepay/api/funding_sources.py

```diff
@@ -0,0 +1,40 @@
+from dataclasses import dataclass
+from typing import List
+
+from littlepay.api import ClientProtocol
+
+
+@dataclass
+class FundingSourceResponse:
+    id: str
+    card_first_digits: str
+    card_last_digits: str
+    card_expiry_month: str
+    card_expiry_year: str
+    card_scheme: str
+    # card_category: str
+    form_factor: str
+    participant_id: str
+    is_fpan: bool
+    related_funding_sources: List[dict]
+    # issuer_country_code: str
+    # issuer_country_numeric_code: str
+    # replacement_funding_source: str
+    # token: str
+    # token_key_id: str
+    # icc_hash: str
```

Comment:
The API lists all these as properties that could show up in a response.

When I manually tested getting a funding source object from the Backoffice API against the QA environment, the JSON response contained only some of these properties. This causes an error about missing keys when we try to [instantiate this class from the response](https://github.com/cal-itp/littlepay/pull/14/files#diff-1a48e786789d311fa76bbddbaa0054032e0b17f742f694ba682a7d46ab0e576cR130), so I commented out the properties that were not in the response.

On the flip side, if we get a JSON response that _does_ have one of the commented out properties, that will throw an error about an "unexpected keyword argument" when we try to instantiate.

I tested with the production environment today, and the JSON response from that _does_ contain some of the commented out properties.

Reply:
Added support in https://github.com/cal-itp/littlepay/pull/14/commits/51d63ec597177b54144698cb0115261e767a5c0f for optional attributes on this dataclass, following suggestion in https://stackoverflow.com/questions/70809438/python-dataclasses-with-optional-attributes